.github/workflows/cicd.yml

```diff
@@ -136,7 +136,7 @@ jobs:
     # Always run this job.
     # Only run this on repos that have self-host runners.
     needs: regular
-    if: github.repository == 'microsoft/ebpf-for-windows'
+    if: github.repository == 'microsoft/ebpf-for-windows' && (github.event_name == 'schedule' || github.event_name == 'pull_request' || github.event_name == 'push' || github.event_name == 'merge_group')
```

Comment:
is this condition equivalent to 'run always except on manual workflow start'?
If so, can you simplify this if statement?

Also, are the perf tests the only ones we would ever run manually?

Reply:
Ah, seeing below it's only the performance_with_profile that is supposed to be run manually, right?

Reply:
1. For my information - can this if-statement be simplified as 'github.event_name != dispatch'?
2. Wondering if we should create a new workflow for this - I see a possibility we want to manually trigger the CICD and have everything run.

Reply:
Maybe. Can we do this as a separate PR?

Reply:
Sure